internal/kafka/internal/services/data_plane_kafka_test.go

```diff
@@ -848,3 +848,130 @@ func Test_DataPlaneKafkaStatus_getStatus(t *testing.T) {
 		})
 	}
 }
+
+func Test_dataPlaneKafkaService_reassignKafkaClusterFull(t *testing.T) {
+	type fields struct {
+		kafkaService   KafkaService
+		clusterService ClusterService
+		kafkaConfig    *config.KafkaConfig
+	}
+	type args struct {
+		kafka *dbapi.KafkaRequest
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "should not return error if success",
+			fields: fields{
+				clusterService: &ClusterServiceMock{
+					FindClusterByIDFunc: func(clusterID string) (*api.Cluster, *errors.ServiceError) {
+						return &api.Cluster{}, nil
+					},
+				},
+				kafkaService: &KafkaServiceMock{
+					GetByIdFunc: func(id string) (*dbapi.KafkaRequest, *errors.ServiceError) {
+						return &dbapi.KafkaRequest{
+							ClusterID:     "test-cluster-id",
+							Status:        constants2.KafkaRequestStatusProvisioning.String(),
+							Routes:        []byte("[{'domain':'test.example.com', 'router':'test.example.com'}]"),
+							RoutesCreated: true,
+						}, nil
+					},
+					UpdatesFunc: func(kafkaRequest *dbapi.KafkaRequest, values map[string]interface{}) *errors.ServiceError {
+						return nil
+					},
+				},
+				kafkaConfig: &defaultKafkaConf,
+			},
+			args: args{
+				kafka: &dbapi.KafkaRequest{
+					Status:    "provisioning",
+					ClusterID: "test-cluster-id",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "should not return error if kafka is not in provisioning status",
```

Comment:
How is this test case different from the above test case?

Reply:
Thanks. Isn't that the same as the first testcase? For that test case to succeed the kafka status has to be in provisioning state.